traffic_ops/traffic_ops_golang/cachegroup/cachegroups.go

```diff
@@ -379,6 +379,23 @@ func (cg *TOCacheGroup) deleteCoordinate(coordinateID int) error {
 	return nil
 }
 
+func GetCacheGroupByName(name string, apiInfo *api.APIInfoImpl) (*tc.CacheGroupNullable, error) {
+	originalParams := apiInfo.ReqInfo.Params
+	apiInfo.ReqInfo.Params = map[string]string{"name": name}
+	cacheGroup := TOCacheGroup{APIInfoImpl: *apiInfo}
+	result, userErr, sysErr, _ := cacheGroup.Read()
+	apiInfo.ReqInfo.Params = originalParams
+	if userErr != nil || sysErr != nil {
+		return nil, util.JoinErrs([]error{userErr, sysErr})
```

Comment:
You're returning potentially sensitive internal errors to clients by doing this.

Reply:
Returning `userErr` only in 4146f5b886.